src/Service/PUDOList.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Answear\DpdPlPickupServicesBundle\Service;
+
+use Answear\DpdPlPickupServicesBundle\ValueObject\Coordinates;
+use Answear\DpdPlPickupServicesBundle\ValueObject\PUDO;
+use GuzzleHttp\Client;
+use GuzzleHttp\ClientInterface;
+
+class PUDOList
+{
+    private ConfigProvider $configProvider;
+    private ClientInterface $client;
+
+    public function __construct(ConfigProvider $configProvider, ?ClientInterface $client = null)
+    {
+        $this->configProvider = $configProvider;
+        $this->client = $client ?? new Client(
+                [
+                    'base_uri' => $configProvider->getUrl(),
+                    'http_errors' => false,
+                    'timeout' => $configProvider->getRequestTimeout(),
+                ]
+            );
```

Comment:
are you sure you want to switch off http_errors?

Reply:
Errors from DPD looks like this:
```
<RESPONSE><ERRORcode="306"><VALUE>Pole max_pudo_number przyjmuje wartości całkowitoliczbowe z przedziału od 1 do 25.</VALUE></ERROR></RESPONSE>
```
and from what I've seen authorization gets 4xx. The idea is that each such response will be turned into an exception. If no proper XML is returned there'll be another exception.